core/shared/src/main/scala/fs2/Segment.scala

```diff
@@ -1431,3 +1432,50 @@ object Segment {
     def defer(t: => Unit): Unit = deferred.addLast(() => t)
   }
 }
+
+trait SegmentInstances {
+  implicit def segmentMonoidInstance[A]: Monoid[Segment[A, Unit]] = new Monoid[Segment[A, Unit]]{
+    def empty: Segment[A, Unit] = Segment.empty[A]
+
+    def combine(x: Segment[A, Unit], y: Segment[A, Unit]): Segment[A, Unit] = x ++ y
+  }
+
+  implicit val defaultSegmentMonadInstance: Traverse[Segment[?, Unit]] with Monad[Segment[?, Unit]]  =
+    new Traverse[Segment[?, Unit]] with Monad[Segment[?, Unit]] {
+      def traverse[G[_], A, B](fa: Segment[A, Unit])(f: (A) => G[B])(implicit evidence$1: Applicative[G]): G[Segment[B, Unit]] =
+        Traverse[List].traverse(fa.force.toList)(f).map(Segment.seq)
+
+      def foldLeft[A, B](fa: Segment[A, Unit], b: B)(f: (B, A) => B): B =
+        fa.force.toList.foldLeft(b)(f)
+
+      def foldRight[A, B](fa: Segment[A, Unit], lb: Eval[B])(f: (A, Eval[B]) => Eval[B]): Eval[B] =
+        Foldable[List].foldRight(fa.force.toList, lb)(f)
+
+      def flatMap[A, B](fa: Segment[A, Unit])(f: (A) => Segment[B, Unit]): Segment[B, Unit] =
+        fa.flatMap(f).mapResult(_ => ())
+
+      def tailRecM[A, B](a: A)(f: (A) => Segment[Either[A, B], Unit]): Segment[B, Unit] =
+        f(a).flatMap[B, Unit]{
+          case Left(l) => tailRecM(l)(f)
+          case Right(r) => Segment(r)
+        }.mapResult(_ => ())
+
+      def pure[A](x: A): Segment[A, Unit] = Segment(x)
+    }
+
+  def segmentRMonad[T]: Monad[Segment[T, ?]] = new Monad[Segment[T, ?]]{
```

Comment:
I'm a bit iffy on this name but I Wasn't sure what to call it...

`rightAppliedSegmentMonad`???